lib/semian/lru_cache.rb

```diff
@@ -0,0 +1,79 @@
+class LRUCache
```

Comment:
Overall I'm finding this linked list stuff hard to follow. It doesn't matter in this prototype but I'd love it if we could find a way to make this easier to understand in the "good copy".

Whether that be through a comment, finding the right abstraction or factoring out of key methods.

Reply:
I've changed the implementation to use a simple hash instead, I used the Linked List stuff from online finding related to LRU implementation with Ruby. The hash gives us the same results and makes it way easier to read.